src/main/java/core/basesyntax/impl/PurchaseHandlerImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.impl;
+
+import dao.FruitDao;
+import dao.FruitDaoImpl;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class PurchaseHandlerImpl implements OperationHandler {
+    private FruitDao fruitDao = new FruitDaoImpl();
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        int quantityBefore = fruitDao.get(transaction.getFruit());
+        if (quantityBefore < transaction.getQuantity()) {
+            throw new RuntimeException("There is not enough fruit in the storage");
+        }
+        fruitDao.add(transaction.getFruit(), quantityBefore - transaction.getQuantity());
```

Comment:
Before saving the updated quantity in the storage, it's important to check that the result balance is positive. If the balance is negative, throw a RuntimeException according to the checklist.